crates/bevy_audio/src/audio_source.rs

```diff
@@ -18,22 +18,92 @@ impl AsRef<[u8]> for AudioSource {
 }
 
 /// Loads mp3 files as [AudioSource] [Assets](bevy_asset::Assets)
+#[cfg(feature = "mp3")]
 #[derive(Default)]
 pub struct Mp3Loader;
 
+#[cfg(feature = "mp3")]
 impl AssetLoader for Mp3Loader {
-    fn load(&self, bytes: &[u8], load_context: &mut LoadContext) -> BoxedFuture<Result<()>> {
-        load_context.set_default_asset(LoadedAsset::new(AudioSource {
-            bytes: bytes.into(),
-        }));
-        Box::pin(async move { Ok(()) })
+    fn load<'a>(
+        &'a self,
+        bytes: &'a [u8],
+        load_context: &'a mut LoadContext,
+    ) -> BoxedFuture<'a, Result<()>> {
+        load(bytes, load_context)
     }
 
     fn extensions(&self) -> &[&str] {
-        &["mp3", "flac", "wav", "ogg"]
+        &["mp3"]
     }
 }
 
+/// Loads flac files as [AudioSource] [Assets](bevy_asset::Assets)
+#[cfg(feature = "flac")]
+#[derive(Default)]
+pub struct FlacLoader;
+
+#[cfg(feature = "flac")]
+impl AssetLoader for FlacLoader {
+    fn load<'a>(
+        &'a self,
+        bytes: &'a [u8],
+        load_context: &'a mut LoadContext,
+    ) -> BoxedFuture<'a, Result<()>> {
+        load(bytes, load_context)
+    }
+
+    fn extensions(&self) -> &[&str] {
+        &["flac"]
+    }
+}
+
+/// Loads wav files as [AudioSource] [Assets](bevy_asset::Assets)
+#[cfg(feature = "wav")]
+#[derive(Default)]
+pub struct WavLoader;
+
+#[cfg(feature = "wav")]
+impl AssetLoader for WavLoader {
+    fn load<'a>(
+        &'a self,
+        bytes: &'a [u8],
+        load_context: &'a mut LoadContext,
+    ) -> BoxedFuture<'a, Result<()>> {
+        load(bytes, load_context)
+    }
+
+    fn extensions(&self) -> &[&str] {
+        &["wav"]
+    }
+}
+
+/// Loads ogg files as [AudioSource] [Assets](bevy_asset::Assets)
+#[cfg(feature = "vorbis")]
+#[derive(Default)]
+pub struct VorbisLoader;
+
+#[cfg(feature = "vorbis")]
+impl AssetLoader for VorbisLoader {
+    fn load<'a>(
+        &'a self,
+        bytes: &'a [u8],
+        load_context: &'a mut LoadContext,
+    ) -> BoxedFuture<'a, Result<()>> {
+        load(bytes, load_context)
+    }
+
+    fn extensions(&self) -> &[&str] {
+        &["ogg"]
+    }
+}
+
+fn load<'a>(bytes: &'a [u8], load_context: &'a mut LoadContext) -> BoxedFuture<'a, Result<()>> {
```

Comment:
Could you give this function a more descriptive name?
`load` is a little bit _overloaded_ (haha) so something like `load_audio_file` or `load_audio_source_bytes` would be a little better.